test/testdrive/privilege_checks.td

```diff
@@ -106,14 +132,155 @@ GRANT CREATE ON CLUSTER sink_cluster TO materialize;
 
 $ postgres-execute connection=mz_system
 REVOKE CREATE, USAGE ON SCHEMA materialize.public FROM materialize;
+REVOKE USAGE ON CONNECTION kafka_conn, csr_conn FROM materialize;
 REVOKE SELECT ON TABLE t FROM materialize;
 REVOKE CREATECLUSTER ON SYSTEM FROM materialize;
 REVOKE CREATE ON CLUSTER sink_cluster FROM materialize;
 DROP SINK s;
 DROP SINK s1;
 DROP TABLE t;
+DROP CLUSTER sink_cluster;
+
+# CREATE CONNECTION
```

Comment:
Creating connections and sources haven't changed in this PR, but I thought adding these tests were relevant and helpful.